### ISSUE TITLE
[JENKINS-70065] Armcc 5.2 warnings not recognized by Armcc parsers

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/analysis-model#1401]
<p>I am having the console log with warnings as below. These are Armcc compiler warning <br/>
But it does not catch the warnings and gives a count as zero.</p>

<pre>
<span class="code-quote">"sw/dbg/cfg/dbg_trace_acfg.h"</span>, line 354: warning #1-D: last line of file ends without a newline
<span class="code-quote">"out/include/common/dimming_ctrl.h"</span>, line 45: warning #1-D: last line of file ends without a newline
</pre>
<p>The command I have used is</p>

<pre>
recordIssues(tools: [armCc()])
</pre>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/krishna_1997.png" title="krishna_1997's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=krishna_1997">krishna_1997</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70065" target="_blank">Armcc 5.2 warnings not recognized by Armcc parsers</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=code_arnab">code_arnab</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: analysis-model
<li><b>label(s)</b>: help-wanted, newbie-friendly
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2025-02-25T19:18:27+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 3
<li><b>imported</b>: 20260108-221235
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I am having the console log with warnings as below. These are Armcc compiler warning 
But it does not catch the warnings and gives a count as zero.</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-quote">"sw/dbg/cfg/dbg_trace_acfg.h"</span>, line 354: warning #1-D: last line of file ends without a newline
<span class="code-quote">"out/include/common/dimming_ctrl.h"</span>, line 45: warning #1-D: last line of file ends without a newline
</pre>
</div></div>
<p>The command I have used is</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">recordIssues(tools: [armCc()])
</pre>
</div></div>

</pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins -	2.319.3.4<br/>
Warnings-ng - 9.7.1
```
</details>
<details><summary><i>1 attachment</i></summary>

- [image-2022-11-11-14-56-50-786.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/59491/image-2022-11-11-14-56-50-786.png)
  > ![image-2022-11-11-14-56-50-786.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/59491/image-2022-11-11-14-56-50-786.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/krishna_1997.png" title="krishna_1997's avatar" /> krishna_1997:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70065?focusedId=432236&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-432236">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/59491/59491_image-2022-11-11-14-56-50-786.png" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p><span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/59491/image-2022-11-11-14-56-50-786.png" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70065?focusedId=432240&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-432240">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Which ARMC compiler version are you using?</p>

<p>The supported formats look like:</p>
<ul class="alternate" type="square">
	<li><a href="https://github.com/jenkinsci/analysis-model/blob/master/src/test/resources/edu/hm/hafner/analysis/parser/armcc5.txt" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model/blob/master/src/test/resources/edu/hm/hafner/analysis/parser/armcc5.txt</a></li>
	<li><a href="https://github.com/jenkinsci/analysis-model/blob/master/src/test/resources/edu/hm/hafner/analysis/parser/armcc.txt" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model/blob/master/src/test/resources/edu/hm/hafner/analysis/parser/armcc.txt</a></li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Which ARMC compiler version are you using?</p>

<p>The supported formats look like:</p>
<ul class="alternate" type="square">
	<li><a href="https://github.com/jenkinsci/analysis-model/blob/master/src/test/resources/edu/hm/hafner/analysis/parser/armcc5.txt" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model/blob/master/src/test/resources/edu/hm/hafner/analysis/parser/armcc5.txt</a></li>
	<li><a href="https://github.com/jenkinsci/analysis-model/blob/master/src/test/resources/edu/hm/hafner/analysis/parser/armcc.txt" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/analysis-model/blob/master/src/test/resources/edu/hm/hafner/analysis/parser/armcc.txt</a></li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/krishna_1997.png" title="krishna_1997's avatar" /> krishna_1997:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70065?focusedId=432273&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-432273">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I am using the armcc compiler </p>

<p>I have provided the format of the warning in the console log in description</p>
</pre>
</details>

</ul>
</details>

<p>I am using the armcc compiler </p>

<p>I have provided the format of the warning in the console log in description</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70065?focusedId=432275&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-432275">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote>
<p>I am using the armcc compiler.</p></blockquote>

<p>But which version?????</p>
</pre>
</details>

</ul>
</details>

<blockquote>
<p>I am using the armcc compiler.</p></blockquote>

<p>But which version?????</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/krishna_1997.png" title="krishna_1997's avatar" /> krishna_1997:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70065?focusedId=432281&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-432281">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Oh sorry </p>

<p>They are using a compiler with the version.</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>armcl version 5.2.4.
</pre>
</div></div>
<p>There is only one arm and we have seen in the supported formats the warning format is the same
 </p>
</pre>
</details>

</ul>
</details>

<p>Oh sorry </p>

<p>They are using a compiler with the version.</p>


```
armcl version 5.2.4.

```
<p>There is only one arm and we have seen in the supported formats the warning format is the same<br/>
 </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>code_arnab:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70065?focusedId=452659&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-452659">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hello <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=drulli" class="user-hover" rel="drulli">drulli</a> I've added a new regex pattern (optional) alongside the original one. This now supports both the new and old warning formats. I will push the PR soon</p>
</pre>
</details>

</ul>
</details>

<p>Hello <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=drulli" class="user-hover" rel="drulli">drulli</a> I've added a new regex pattern (optional) alongside the original one. This now supports both the new and old warning formats. I will push the PR soon</p>
